### ISSUE TITLE
os/gcfg: priority of env should be higer than file.

### ISSUE THREAD

**Issue by yangyanqing** [gogf/gf#4074]
### Go version

go version go1.23.4 windows/amd64

### GoFrame version

2.8.3

### Can this bug be reproduced with the latest release?

Option Yes

### What did you do?

With v2.8.3, config by file had higher than config by env.

### What did you see happen?

```go
func (c *Config) Get(ctx context.Context, pattern string, def ...interface{}) (*gvar.Var, error) {
	var (
		err   error
		value interface{}
	)
	value, err = c.adapter.Get(ctx, pattern)
	if err != nil {
		return nil, err
	}
	if value == nil {
		if len(def) > 0 {
			return gvar.New(def[0]), nil
		}
		return nil, nil
	}
	return gvar.New(value), nil
}
```

### What did you expect to see?

In general, config by env should have higher priority than config by config.yaml.
Spring cloud and viper think so too.

**Comment by gqcn** [comment]
@yangyanqing Yes, I think so too. It should be updated in method `GetWithEnv` and `GetWithCmd` for package `gcfg`:
- https://github.com/gogf/gf/blob/6ea1526b751f7ec5b466eb98016402748eb793d5/os/gcfg/gcfg.go#L121
- https://github.com/gogf/gf/blob/6ea1526b751f7ec5b466eb98016402748eb793d5/os/gcfg/gcfg.go#L143

Thanks for your proposal!

**Comment by lingcoder** [comment]
PR #4587 的改动是一个破坏性变更，不应该在小版本中引入。很多生产环境的服务依赖原有的优先级行为（配置文件 > 环境变量），这个改动会导致现有项目行为异常。 已提交 PR #4647 恢复原有行为。如果需要"环境变量 > 配置文件"的优先级，建议后续通过新增方法来支持，而不是修改现有方法的行为。

**Comment by yangyanqing** [comment]
> PR [#4587](https://github.com/gogf/gf/pull/4587) 的改动是一个破坏性变更，不应该在小版本中引入。很多生产环境的服务依赖原有的优先级行为（配置文件 > 环境变量），这个改动会导致现有项目行为异常。 已提交 PR [#4647](https://github.com/gogf/gf/pull/4647) 恢复原有行为。如果需要"环境变量 > 配置文件"的优先级，建议后续通过新增方法来支持，而不是修改现有方法的行为。

除了 gf，还没发现哪个工具中的 配置文件优先级高于环境变量 @lingcoder

**Comment by lingcoder** [comment]
> > PR [#4587](https://github.com/gogf/gf/pull/4587) 的改动是一个破坏性变更，不应该在小版本中引入。很多生产环境的服务依赖原有的优先级行为（配置文件 > 环境变量），这个改动会导致现有项目行为异常。 已提交 PR [#4647](https://github.com/gogf/gf/pull/4647) 恢复原有行为。如果需要"环境变量 > 配置文件"的优先级，建议后续通过新增方法来支持，而不是修改现有方法的行为。
> 
> 除了 gf，还没发现哪个工具中的 配置文件优先级高于环境变量 [@lingcoder](https://github.com/lingcoder)

你说的没错，但是这个优先级已经存在好几年了。突然改变会直接影响现有生产项目。直接掰正会导致大量线上用户的配置读取结果改变。
